install-script/install.ps1

```diff
@@ -396,19 +404,48 @@ function Get-BinaryFromUri {
     Write-Host "Downloaded ${Path}"
 }
 
+function Send-Installation-Logs {
+    param (
+        [Parameter(Mandatory, Position=0)]
+        [HttpClient] $HttpClient,
+
+        [Parameter(Mandatory, Position=1)]
+        [string] $Endpoint,
+
+        [Parameter(Mandatory, Position=2)]
+        [string] $Path
+    )
+
+    $Content = Get-Content -Path $Path
+    $StringContent = [System.Net.Http.StringContent]::new($Content)
+
+    $response = $HttpClient.PostAsync($Endpoint, $StringContent).GetAwaiter().GetResult()
+    if (!($response.IsSuccessStatusCode)) {
+        $statusCode = [int]$response.StatusCode
+        $reasonPhrase = $response.StatusCode.ToString()
+        $errMsg = "${statusCode} ${reasonPhrase}"
+
+        if ($response.Content -ne $null) {
+            $content = $response.Content.ReadAsStringAsync().GetAwaiter().GetResult()
+            $errMsg += ": ${content}"
+        }
+
+        Write-Error $errMsg -ErrorAction Stop
+    }
+}
+
 ##
 # Main code
 ##
 
 try {
-    if ($InstallationToken -eq $null -or $InstallationToken -eq "") {
-        Write-Error "Installation token has not been provided. Please set the SUMOLOGIC_INSTALLATION_TOKEN environment variable." -ErrorAction Stop
+    $InstallationLogFile = New-TemporaryFile
+
+    if ($InstallationLogFileEndpoint -eq "") {
+        $InstallationLogFileEndpoint = "https://stag-open-events.sumologic.net/api/v1/collector/installation/logs"
```

Comment:
This endpoint is not publicly accessible, as far as I know. Is it really what we should be using here?

Reply:
The endpoint is actually publicly accessible (be it for staging or production), else we wouldn't be able to send installation logs to it from wherever the collector gets installed. The URL is currently set to the staging endpoint, but I'll change it to the production one.

The backend team went through a separate security review and subsequent changes to make sure that this endpoint was OK to have as publicly accessible.